### PR TITLE
fix(Subs): distinguish visual block mode and visual line mode

### PR DESCRIPTION
**Pull request by peterfication**
Until now, it was only working properly for visual block mode. If visual
line had a short last line selected, it wouldn't work because in visual
block mode, this means that the range is as short as the last line.

So if the last line was a line with 0 chars, which can happen easily
in coding files, it would not work.

Example:

```typescript
const foo = "bar";
const bar = "bar";

console.log(foo);
````

In this example, selecting the first two lines and doing a `Subs` will
work as expected. However, if the third line is selected, nothing will
happen because the third line has 0 chars and hence the visual block
mode will have a range of 0 chars.

**This change fixes that.**

Fixes https://github.com/johmsalas/text-case.nvim/issues/117

### CODE REVIEW

**Comment by johmsalas** [comment]
TDD 👍

File: lua/textcase/plugin/plugin.lua [hunk 1]
```diff
@@ -135,17 +135,28 @@ end
 -- <cmd>h command-preview
 function M.incremental_substitute(opts, preview_ns, preview_buf)
   local command = "Subs"
+
+  -- preview_ns and preview_buf indicates the buffer to be modified
```
Comment (johmsalas) [inline lua/textcase/plugin/plugin.lua:139]: This is clever

**Review by johmsalas — APPROVED** [review]
I have been trackings similar issues about the difference between V and v from the early days of this plugin and hadn't noticed a relation with the line lenght.

An example of this struggle is the first 4 methods i[n the utils file](https://github.com/johmsalas/text-case.nvim/blob/97b8fc077080d109114f409d25bc0291b4375817/lua/textcase/shared/utils.lua#L4) are dedicated to the visual region

File: lua/textcase/plugin/plugin.lua (hunk 1 above)
Comment (peterfication) [reply lua/textcase/plugin/plugin.lua:139]: This was there before from you ;) It just moved up a bit because I needed the buffer for my hack as well ;)

**Pull request merged**